File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> s(nums.begin(), nums.end());
        int longest = 0;

        for (int num : s) {                // iterate over the set (not nums)
            if (!s.count(num - 1)) {       // only start from the beginning of a run
                int curr = num;
                int length = 1;

                while (s.count(curr + 1)) {
                    curr++;
                    length++;
                }

                longest = max(longest, length);
            }
        }

        return longest;
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function (nums) {
    if(nums.length == 0) return 0;
    nums.sort((a, b) => a - b);
    console.log(nums);
    let max = 1;
    let curr = 1;
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] == nums[i - 1]) continue; // duplicated
        if (i > 0 && nums[i - 1] == nums[i] - 1) {
            curr += 1;
            max = Math.max(max, curr);
        } else {
            curr = 1;
        }
    }
    return max;
};